gio/src/input_stream.rs

```diff
@@ -525,12 +526,27 @@ impl<T: IsA<InputStream>> InputStreamAsyncBufRead<T> {
     }
 }
 
-#[derive(thiserror::Error, Debug)]
+#[derive(Debug)]
 enum BufReadError {
-    #[error("Previous read operation failed")]
     Failed,
 }
 
+impl std::error::Error for BufReadError {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        match self {
+            Self::Failed => None,
```

Comment:
This makes no sense, that's the default implementation

Reply:
No, this is needed.

Reply:
Ok nevermind, I didn't realize I removed the `derive(Error)`...